### PR TITLE
Issue 250

### PR DESCRIPTION
**Pull request by grgrzybek**

### CODE REVIEW

**Comment by tadayosi** [comment]
Fix #250

**Review by tadayosi — COMMENTED** [review]
Hmm, it's not clear to me how the fix addresses the problem stated by Roland:
https://github.com/jolokia/jolokia/issues/250#issuecomment-206513284
> Unfortunately this is currently not possible since the request is [added always to the response](https://github.com/rhuss/jolokia/blob/master/agent/core/src/main/java/org/jolokia/backend/BackendManager.java#L431-L431), which is use for correlation e.g. in bulk request.

Maybe you can add a test for bulk requests with this option to verify it's working with them.

**Comment by grgrzybek** [comment]
I can add a test showing it doesn't simply work ;)
I will mention more explicitly that if you turn this option on, you have to correlate using different method (by order of responses).

**Comment by tadayosi** [comment]
@grgrzybek Yes, I think it's very important as bulk request is a quite common method to optimise Jolokia requests.

**Comment by grgrzybek** [comment]
Oh - I wrote a test which failed... I wanted to ensure that with `includeRequest=false` we're not getting the request inside the response. But it was there!

The point is that `org.jolokia.client.request.ValidatingResponseExtractor#extract()` method is used at client side and it's actually putting the request back into the response - whether or not the response contains its request...

So this confirms my suspicion that we can correlate simply by iterating requests and responses in parallel.

File: src/documentation/manual/modules/ROOT/pages/client/java.adoc [hunk 1]
```diff
@@ -360,6 +360,13 @@ they are created. The allowed values are either
 order (== alphabetical sorted) is used or
 `false` for getting the keys as
 registered. Default is `true`
+
+|`INCLUDE_REQUEST`
+|Whether the response object should contain related request object. +
+This option may be configured globally and overriden at request time.
+When `true`, bulk responses have to be correlated with requests by matching
```
Comment (tadayosi) [inline src/documentation/manual/modules/ROOT/pages/client/java.adoc:367]: Maybe it means `false` here?

File: src/documentation/manual/modules/ROOT/pages/client/javascript.adoc [hunk 2]
```diff
@@ -251,6 +251,13 @@ reads.
 |`canonicalNaming`
 |Defaults to true for canonical (sorted) property lists on object names; if set to "false" then they are turned in their unsorted format.
 
+|`includeRequest`
+|Whether the response object should contain related request object. +
+This option may be configured globally and overriden at request time.
+When `true`, bulk responses have to be correlated with requests by matching
```
Comment (tadayosi) [inline src/documentation/manual/modules/ROOT/pages/client/javascript.adoc:257]: `true` -> `false`

File: src/documentation/manual/modules/ROOT/pages/jolokia_protocol.adoc [hunk 3]
```diff
@@ -561,7 +561,10 @@ Only `text/plain` and `application/json` are allowed.
 For any other value Jolokia will fallback to `text/plain`.
 
 `includeRequest`:: A flag (defaults to `true`) which controls whether the incoming
-request should be included in the response (under `request` key).
+request should be included in the response (under `request` key). +
+When `true`, bulk responses have to be correlated with requests by matching
```
Comment (tadayosi) [inline src/documentation/manual/modules/ROOT/pages/jolokia_protocol.adoc:565]: `true` -> `false`

**Review by tadayosi — APPROVED** [review]
Some minor typos but otherwise LGTM.

File: src/documentation/manual/modules/ROOT/pages/jolokia_protocol.adoc (hunk 3 above)
Comment (grgrzybek) [reply src/documentation/manual/modules/ROOT/pages/jolokia_protocol.adoc]: thanks for detailed review!

**Pull request merged**